venom/rpc/reflect/openapi.py

```diff
@@ -1,6 +1,195 @@
+from itertools import groupby
+from functools import singledispatch
+from collections import defaultdict
 from venom.rpc.reflect.reflect import Reflect
-from venom.rpc.reflect.stubs import OpenAPISchema
+from venom.rpc.reflect.stubs import TypeMessage, MethodMessage, \
+    ParameterMessage, ResponsesMessage, ResponseMessage, \
+    FieldsMessage, InfoMessage, OpenAPISchema
+from venom.protocol import JSON
+from venom.fields import Field, RepeatField, MapField
+from venom import Message
+from venom.message import field_names, fields
+from venom.rpc.method import Method, HTTPFieldLocation
+
+
+DEFINITIONS = 'definitions'
+DESCRIPTION = 'description'
+MESSAGE = 'message'
+ARRAY = 'array'
+MAP = 'map'
+
+
+# TODO: other types
+TYPES = {
+    'str': 'string',
+    'int': 'integer',
+    'float': 'double',
+    'bool': 'boolean',
+}
+
+PATH_PARAMETER = {
+    'is_in': 'path',
+    'required': True
+}
+
+BODY_PARAMETER = {
+    'is_in': 'body',
+}
+
+QUERY_PARAMETER = {
+    'is_in': 'query',
+}
+
+
+def description_param(field: Field) -> dict:
+    if DESCRIPTION in field.options:
+        return {DESCRIPTION: field.options.get(DESCRIPTION)}
+    return {}
+
+
+@singledispatch
+def type_message(field: Field) -> TypeMessage:
+    f_type = field.type.__name__
+    if f_type not in TYPES:
+        return TypeMessage(ref=reference_string(field.type))
+    return TypeMessage(
+        type=TYPES[f_type],
+        **description_param(field)
+    )
+
+
+@type_message.register(RepeatField)
+def type_message_repeat(field: RepeatField) -> TypeMessage:
+    return TypeMessage(
+        type=ARRAY,
+        items=type_message(field.items),
+        **description_param(field)
+    )
+
+
+@type_message.register(MapField)
+def type_message_map(field: MapField) -> TypeMessage:
+    return TypeMessage(
+        type='object',
+        additionalProperties=type_message(field.values),
+        **description_param(field)
+    )
+
+
+def reference_string(message: Message) -> str:
+    name = message.__meta__.name
+    if name == 'Empty':
```

Comment:
This is hacky. First, that is not the right way to test that `message` is an instance of `Empty`. Also not only `Empty` is guaranteed to be empty. Other messages can be empty too.